lib/components/hikingmapbox.dart

```diff
@@ -167,30 +168,66 @@ class _MapWidgetState extends State<MapWidget> {
   }
 
   void updateRoute() async {
-    List<LatLng> remainingRoute = [];
-    List<LatLng> passedRoute = _passedRoute;
-
-    for (int i = 0; i < _route.length; i++) {
-      LatLng latLng = _route[i];
-      double distanceToCurrentLocation =
-          OsmData.getDistance(latLng, _currentDeviceLocation);
-      if (distanceToCurrentLocation > 0.0002) {
-        remainingRoute.add(latLng);
-      } else {
-        passedRoute.add(latLng);
+    int deleteAllRouteNodesWithAnIndexSmallerThan = 0;
+    for (int index = 0; index < _route.length; index++) {
+      if (shouldRouteNodeAtIndexBeConsideredForRemoval(index)) {
+        double distanceToCurrentLocation =
+            OsmData.getDistance(_route[index], _currentDeviceLocation);
+        if (distanceToCurrentLocation < 0.05) {
+          deleteAllRouteNodesWithAnIndexSmallerThan = index + 1;
+        }
       }
     }
 
+    List<LatLng> remainingRoute =
+        _route.sublist(deleteAllRouteNodesWithAnIndexSmallerThan);
+
     LineOptions optionsRemainingRoute = LineOptions(geometry: remainingRoute);
     await mapController.updateLine(_lineRoute, optionsRemainingRoute);
 
+    List<LatLng> passedRoute = [
+      ..._passedRoute,
+      ..._route.sublist(0, deleteAllRouteNodesWithAnIndexSmallerThan + 1)
+    ];
     LineOptions optionsPassedRoute = LineOptions(geometry: passedRoute);
     await mapController.updateLine(_linePassedRoute, optionsPassedRoute);
 
     setState(() {
       _route = remainingRoute;
       _passedRoute = passedRoute;
     });
+
+    if(_route.length <= 1){
+      finishHikingTrip();
+    }
+
+  }
+
+  bool shouldRouteNodeAtIndexBeConsideredForRemoval(int index) {
```

Comment:
again this method naming is extremly long and does not specify what the method really does, only how it is used i guess.
What happens when the route is constructed out of 51 nodes? all route nodes will never be considered for removal. I think that is not the goal you are aiming for here or am i wrong?